uiowa/uiowa
# [Registrar] five year calendar

Resolves https://github.com/uiowa/uiowa/issues/4943. 



# How to test

```
ddev blt ds --site=registrar.uiowa.edu && ddev drush @registrar.local uli /calendars-and-deadlines/five-year-calendar
```

- Add the "Five Year Calendar" block to the page.
- Confirm that it defaults to current academic year.
- Confirm that each year is listed in this order:
  -  Fall
  -  Winter
  -  Spring
  -  Summer
    -  4wk
    -  6wk I
    -  6wk II
    -  8wk 
    -  12wk
- Compare events against https://registrar.uiowa.edu/five-year-calendar and confirm same events are being shown.
- Confirm url hash is updated each time the Academic Year select changes. 
- Confirm reset button resets to default state.
- Look at Controller refactor and confirm that https://registrar.uiowa.ddev.site/calendars-and-deadlines/academic-calendar is still working as expected. 
- Code review 
  - Please feel free to change or suggest changes to make the code better!
